### ISSUE TITLE
Non-optimal if conditions false positive

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#670]
```php
if (!isset($params[$name]) && !array_key_exists($name, $params)) {
    //
}
```
`array_key_exists` is marked, but it costs more than `isset`.

**Comment by rentalhost** [comment]
Why are you checking if `isset($params[$number])` is not defined, then checking if the key do not exists too? It is not redundant here? If first is true, the second will be too.

**Comment by b1rdex** [comment]
@rentalhost that's because `isset(null) === false`, but `array_key_exists` returns `true` for such array values.

**Comment by funivan** [comment]
Maybe you can just remove `isset` ?

**Comment by b1rdex** [comment]
Sure I can. But AFAIK isset is much faster for big arrays.

12 дек. 2017 г. 6:17 ПП пользователь "Ivan" <notifications@github.com>
написал:

**Comment by b1rdex** [comment]
Anyway isset is language operator and it's faster than function. So this is
false positive.

12 дек. 2017 г. 6:18 ПП пользователь "Anatoly Pashin" <
anatoly.pashin@gmail.com> написал:

**Comment by funivan** [comment]
@b1rdex agree. `isset` is faster ;)
